File: program23.c
#include <stdio.h>

void activitySelection(int start[], int finish[], int n) {
    int i, j;
    printf("Selected activities are: ");
    
    i = 0; // first activity always selected
    printf("%d ", i + 1);

    for (j = 1; j < n; j++) {
        if (start[j] >= finish[i]) {
            printf("%d ", j + 1);
            i = j;
        }
    }
    printf("\n");
}

int main() {
    int n, i;
    printf("Enter number of activities: ");
    scanf("%d", &n);

    int start[n], finish[n];
    printf("Enter start times: ");
    for (i = 0; i < n; i++)
        scanf("%d", &start[i]);

    printf("Enter finish times: ");
    for (i = 0; i < n; i++)
        scanf("%d", &finish[i]);

    activitySelection(start, finish, n);
    return 0;
}

File: program23.py
# Activity Selection Problem using Greedy Algorithm

def activity_selection(start, finish):
    n = len(finish)

    # Sort activities by finish time
    activities = sorted(zip(start, finish), key=lambda x: x[1])

    selected = [activities[0]]  # Always select first activity

    # Select subsequent activities
    for i in range(1, n):
        if activities[i][0] >= selected[-1][1]:
            selected.append(activities[i])

    return selected


# ----- MAIN PROGRAM -----
print("Activity Selection Problem using Greedy Algorithm\n")

n = int(input("Enter number of activities: "))

start = list(map(int, input("Enter start times : ").split()))
finish = list(map(int, input("Enter finish times: ").split()))

selected = activity_selection(start, finish)

print("\nSelected activities:")
for s, f in selected:
    print(f"Start: {s}, Finish: {f}")
